indico/modules/users/blueprint.py

```diff
@@ -56,6 +57,7 @@
     _bp.add_url_rule('/emails/verify/<token>', 'user_emails_verify', RHUserEmailsVerify)
     _bp.add_url_rule('/emails/<email>', 'user_emails_delete', RHUserEmailsDelete, methods=('DELETE',))
     _bp.add_url_rule('/emails/make-primary', 'user_emails_set_primary', RHUserEmailsSetPrimary, methods=('POST',))
+    _bp.add_url_rule('/block/', 'user_block', RHUserBlock, methods=('PUT', 'DELETE'))
```

Comment:
I can omit the trailing slash but it will become the only entry of this block without one.

Reply:
Not sure about the `/blocked` though. In theory, we will PUT to `block` and DELETE a `block`. The same with the previous `/emails/make-primary`, we will POST to "make primary".

I think it makes more sense to have the route reflecting an action we will trigger in the present, and not in the past.

Reply:
there's nothing "inside" `/block/`, that's why i'd skip the slash (`/emails/` has `/emails/<email>` inside for example)

Reply:
the idea behind PUTting and DELETEing `/user/blocked` is that you are setting/clearing the "blocked" flag on the user, while you POST the make-primary verb (even though this isn't fully restful - usually one would PATCH the email with `{"is_primary": true}` to be fully restful)

Reply:
> there's nothing "inside" `/block/`, that's why i'd skip the slash (`/emails/` has `/emails/<email>` inside for example)

Should we keep the following without it?
```
_bp.add_url_rule('/profile/', 'user_profile', RHPersonalData, methods=('GET', 'POST'))
_bp.add_url_rule('/preferences/', 'user_preferences', RHUserPreferences, methods=('GET', 'POST'))
_bp.add_url_rule('/favorites/', 'user_favorites', RHUserFavorites)
```

Reply:
those are user-facing so i wouldn't change them since there's no redirect from slash to no-slash, only vice versa (but we should probably have done them without the slash in the first place).

anyway, the block action will only be called in the background and never show up for an end user

Reply:
I think the crux here is that we're trying to approach the ideal PATCH solution. I agree with both solutions even though `block` initially seemed more straight-forward when going against the pattern.

I've changed it to `blocked`.